lib/betagouvbot/webhook.rb

```diff
@@ -95,5 +101,23 @@ def members
     error StandardError do
       "Zut, il y a une erreur: #{env['sinatra.error'].message}"
     end
+
+    ## Error handling
+
+    error ArgumentError,
+          AccountRequest::InvalidNameError,
+          AccountRequest::InvalidEmailError do
+
+      error_message do |response|
```

Comment:
There're several points to your comments.

1. We should always give feedback to our users. Coming from an API, we should grant a code 4xx, an error type and an a message explaining the user how to change the way it is calling the endpoint (or if it's a 500 a message inviting them to retry or to ping us on Slack). If the end-user is on Slack (he won't see our message with the /slash commands, BTW I think we should rather use the newer API), then we need to make our best to deliver them an explanation.

2. Using exceptions for control flow is like an ice cream : you either love it or you don't, which doesn't mean it is good or bad. We can go ask our data.gouv.fr friends and you'll see their position about control flow with exceptions is rather positive.

3. We're not doing control flow with exceptions, we're just notifying the user something went wrong. Task execution is halted immediately after exception is raised.

4. In terms of maintenance, you know that rescuing from exceptions or not, you shall not answer with a 2xx if there was an error. With proper error types, messages and specs, you'll never have a obscuring control flow problem (neither need for debugging).

5. Finally the point on avoiding short-circuiting is effectively interesting while for example validating input data. Would you be kind enough to propose a rough implementation of that idea of yours?